File: GXpert/GXpert.Web/Modules/Masters/PreAcademicScore/PreAcademicScoreColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Masters.Columns;

[ColumnsScript("Masters.PreAcademicScore")]
[BasedOnRow(typeof(PreAcademicScoreRow), CheckNames = true)]
public class PreAcademicScoreColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public string PreAcadamicsExamName { get; set; }
    public DateTime PassedOutDate { get; set; }
    public float MarksObtained { get; set; }
    public float OutOfMarks { get; set; }
    public string StudentPrn { get; set; }
    [EditLink]
    public string Remarks { get; set; }
   
}

File: GXpert/GXpert.Web/Modules/ServerTypes/Masters/PreAcademicScoreColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { PreAcademicScoreRow } from "./PreAcademicScoreRow";

export interface PreAcademicScoreColumns {
    Id: Column<PreAcademicScoreRow>;
    PreAcadamicsExamName: Column<PreAcademicScoreRow>;
    PassedOutDate: Column<PreAcademicScoreRow>;
    MarksObtained: Column<PreAcademicScoreRow>;
    OutOfMarks: Column<PreAcademicScoreRow>;
    StudentPrn: Column<PreAcademicScoreRow>;
    Remarks: Column<PreAcademicScoreRow>;
}

export class PreAcademicScoreColumns extends ColumnsBase<PreAcademicScoreRow> {
    static readonly columnsKey = 'Masters.PreAcademicScore';
    static readonly Fields = fieldsProxy<PreAcademicScoreColumns>();
}